Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -123,30 +586,80 @@ class GeneralForceSubsystemRep : public ForceSubsystem::Guts {
 
         // Some forces are disabled by default; initialize the enabled flags
         // accordingly. Also, see if we're going to need to do any caching
-        // on behalf of any forces that don't depend on velocities. 
+        // on behalf of any forces that don't depend on velocities.
         Array_<bool> forceEnabled(getNumForces());
         bool someForceElementNeedsCaching = false;
         for (int i = 0; i < (int)forces.size(); ++i) {
             forceEnabled[i] = !(forces[i]->isDisabledByDefault());
             if (!someForceElementNeedsCaching)
-                someForceElementNeedsCaching = 
+                someForceElementNeedsCaching =
                     forces[i]->getImpl().dependsOnlyOnPositions();
         }
 
-        forceEnabledIndex = allocateDiscreteVariable(s, Stage::Instance, 
+        forceEnabledIndex = allocateDiscreteVariable(s, Stage::Instance,
             new Value<Array_<bool> >(forceEnabled));
+        
+        
+        // Track the enabled forces and whether they should be parallelized.
+        Array_<Force*> enabledNonParallelForces;
+        Array_<Force*> enabledParallelForces;
+
+        // Avoid repeatedly allocating memory.
+        enabledNonParallelForces.reserve(forces.size());
+        enabledParallelForces.reserve(forces.size());
 
-        // Note that we'll allocate these even if all the needs-caching 
+        for (int i = 0; i < (int) forces.size(); ++i) {
+            if(forceEnabled[i])
+            {
+                if (forces[i]->getImpl().shouldBeParallelIfPossible())
+                    enabledParallelForces.push_back(forces[i]);
+                else
+                    enabledNonParallelForces.push_back(forces[i]);
+            }
+        }
+
+        enabledNonParallelForcesIndex = allocateCacheEntry(s, Stage::Instance,
+                          new Value<Array_<Force*> >(enabledNonParallelForces));
+        enabledParallelForcesIndex = allocateCacheEntry(s, Stage::Instance,
+                             new Value<Array_<Force*> >(enabledParallelForces));
+
+        //Determine whether the subsystem has parallel forces - if so, use the
+        //parallel implementation of CalcForcesTask (even if those parallel
+        //forces are not currently enabled - they could be enabled in the
+        //future)
+        //
+        // *Consider the following example:*
+        // Parallel Forces: A B C
+        // NonParallel Forces: D E
+        //
+        // Enabled Forces at Time 0: D E
+        // Enabled Forces at Time 1: A B C D E
+        
+        bool hasParallelForces = false;
+        for(int x = 0; x < forces.size(); ++x)
```

Comment:
This line gives a signed/unsigned mismatch warning in Visual Studio. You can use:
```c++
for(int x = 0; x < (int)forces.size(); ++x)
```
or use an `unsigned` iteration variable.